arch/arm/soc/atmel_sam/same70/soc.h

```diff
@@ -46,6 +46,10 @@
 #include "../common/soc_pmc.h"
 #include "../common/soc_gpio.h"
 
+/* ARM CMSIS definitions must be included before kernel_includes.h.
+ * Therefore, it is essential to include kernel_includes.h after including
+ * core SOC-specific headers.
+ */
```

Comment:
It would be nice to have it documented (in the commit message or comment) why CMSIS definitions must be included before kernel_includes.h.

Reply:
Yeap, I can update the commit message with some content taken from the explanatory comment.

Reply:
@SebastianBoe : updated the commit message